umbraco/Umbraco-CMS
# V9: Add basic auth middleware

## Notes
Adds a middleware, that allows for prompting for basic auth on the website part if:
- The basic auth is turned on
- The IP is not allow listed
- No valid backoffice cookie exists


## Details
- Added new settings under `Umbraco:CMS:BasicAuth`
- Added service that caches the settings, and updates the cache on changes
- Added middleware that executes as the first middleware (by default)



## Tests
- Ensure you are not prompted when disabled (Default)
- Turn on basic auth in config `Umbraco:CMS:BasicAuth:Enabled = true`
- Go to the website when not logged in in the backoffice.
   - You will now be prompted using basic auth
   - Type backoffice credentials
 - Ensure you are not prompted again on future requests.
 
**Note that, browsers cache the basic auth credentials, so if you login using basic auth, and go to backoffice and logout, you will automatically be signed in on next website request.**
